### DISCUSSION TITLE
Large Scale Area Selection Analysis

### DISCUSSION THREAD

**Discussion by tehfireburnz** [maruohon/litematica#759] [Ideas]
When using an area selection of larger than 2x render distance, it's difficult to keep the analysis window open (even though there is scheduled task for this. It would be nice if the analysis display featured similar to the schematic verifier where you could open it from cache without having to then "re-analyze" the entire selection every time you want to look at it.

Another feature that would be welcomed similar to the verifier would be able to select the block type and it highlight those specific blocks within your area selection.

For reference this issue has spawned from a very large perimeter with a less-than optimal bombing strategy which has left some stray blocks which are difficult to identify. As a workaround currently I'm using much smaller area-selections to allow me to detect stray blocks, however without block highlights it's still difficult.

Do you think this is something you would consider implementing and others would deem valuable?

On a side note, I suppose I could create a completely "null" schematic with just air in it and use the verifier that way :)

**Comment by maruohon** [comment]
I think you can open the last area analyzer result again by using the material list GUI hotkey, because it uses the Material List system currently.

I know all of this is really janky atm, and the entire Material List system will get a ton of changes and improvements at some point. And additionally the Area Analyzer will be rewritten as a separate system, so that it doesn't use the Material List system anymore. The Material List system needs to convert all the blocks to items, which means that the analyzer won't even show you blocks that don't have an associated item at all, plus the block-to-item association isn't always a direct 1:1 mapping. So having it just count and display the block states directly would be a lot better for the analyzer.

Highlighting blocks from the area will probably also get added when the system is rewritten at some point. I know these features will be a concern on some/most non-technical(?) servers for obvious cheating possibilities, so these things will be controllable from the server side via the new malilib config locking and action disable capabilities in the newer code base.
